wandb/sdk/launch/runner/aws.py

```diff
@@ -0,0 +1,374 @@
+import configparser
+import itertools
+import logging
+import os
+import subprocess
+import time
+from typing import Any, Dict, List, Optional, Tuple
+
+if False:
+    import boto3  # type: ignore
+import wandb
+import wandb.docker as docker
+from wandb.errors import CommError, LaunchError
+from wandb.util import get_module
+
+from .abstract import AbstractRun, AbstractRunner, Status
+from .._project_spec import (
+    get_entry_point_command,
+    LaunchProject,
+)
+from ..docker import (
+    build_docker_image_if_needed,
+    construct_local_image_uri,
+    docker_image_exists,
+    docker_image_inspect,
+    generate_docker_base_image,
+    pull_docker_image,
+    validate_docker_installation,
+)
+from ..utils import PROJECT_DOCKER_ARGS
+
+
+_logger = logging.getLogger(__name__)
+
+
+class SagemakerSubmittedRun(AbstractRun):
+    """Instance of ``AbstractRun`` corresponding to a subprocess launched to run an entry point command on aws sagemaker."""
+
+    def __init__(self, training_job_name: str, client: "boto3.Client") -> None:
+        super().__init__()
+        self.client = client
+        self.training_job_name = training_job_name
+        self._status = Status("running")
+
+    @property
+    def id(self) -> str:
+        return f"sagemaker-{self.training_job_name}"
+
+    def wait(self) -> bool:
+        while True:
+            status_state = self.get_status().state
+            if status_state in ["stopped", "failed", "finished"]:
+                break
+            time.sleep(5)
+        return status_state == "finished"
+
+    def cancel(self) -> None:
+        # Interrupt child process if it hasn't already exited
+        status = self.get_status()
+        if status.state == "running":
+            self.client.stop_training_job(TrainingJobName=self.training_job_name)
+            self.wait()
+
+    def get_status(self) -> Status:
+        job_status = self.client.describe_training_job(
+            TrainingJobName=self.training_job_name
+        )["TrainingJobStatus"]
+        if job_status == "Completed":
+            self._status = Status("finished")
+        elif job_status == "Failed":
+            self._status = Status("failed")
+        elif job_status == "Stopping":
+            self._status = Status("stopping")
+        elif job_status == "Stopped":
+            self._status = Status("finished")
```

Comment:
could we combine the two `Status("finished")` lines into one if statement just for clarity

Reply:
Absolutely